### PR TITLE
Prohibit some Random usage patterns

### PR DESCRIPTION
**Pull request by leventov**
- `Math.abs(random.nextInt())` doesn't guarantee positivity
- `random.nextInt() % x`
- `new Random()` should never be used, `ThreadLocalRandom.current()` or `SplittableRandom` instead.

I also replaced most `new Random(SOME_CONSTANT_SEED)` in benchmarks with `ThreadLocalRandom.current()` because benchmarks shouldn't generally be always identical, randomness between the runs could make results statistically more reliable.

### CODE REVIEW

**Comment by jon-wei** [comment]
@leventov Can you fix conflicts?

**Comment by leventov** [comment]
@jon-wei fixed

File: indexing-service/src/main/java/org/apache/druid/indexing/common/task/RealtimeIndexTask.java [hunk 1]
```diff
@@ -76,34 +76,43 @@
...
 public class RealtimeIndexTask extends AbstractTask
 {
   public static final String CTX_KEY_LOOKUP_TIER = "lookupTier";
 
   private static final EmittingLogger log = new EmittingLogger(RealtimeIndexTask.class);
-  private static final Random random = new Random();
+
+  private static final int TASK_ID_BITS_PER_SYMBOL = 4;
+  private static final int TASK_ID_SYMBOL_MASK = (1 << TASK_ID_BITS_PER_SYMBOL) - 1;
+  private static final int TASK_ID_LENGTH = Integer.BYTES / TASK_ID_BITS_PER_SYMBOL;
```
[the remark this answers is not in the document]
Comment (jon-wei) [reply indexing-service/src/main/java/org/apache/druid/indexing/common/task/RealtimeIndexTask.java]: @glasser This was fixed in #6383

[the remark this answers is not in the document]
Comment (glasser) [reply indexing-service/src/main/java/org/apache/druid/indexing/common/task/RealtimeIndexTask.java]: I guess that'll teach me to run off of a random commit on master :)

**Pull request merged**